### PR TITLE
Switch to github actions for CI

### PR DESCRIPTION
**Pull request by djhoese**

### CODE REVIEW

File: .github/workflows/deploy-sdist.yaml
Comment (pearu) [reply .github/workflows/deploy-sdist.yaml:25]: Done, @djhoese

Comment (djhoese) [reply .github/workflows/deploy-sdist.yaml:25]: Awesome! Thanks!

**Comment by djhoese** [comment]
FYI turns out there is a decent chunk of functionality that isn't supported on OSX:

```
 FAILED libtiff/tests/test_tiff_array.py::test_simple_slicing - SystemError: mmap: resizing not available--no mremap()
FAILED libtiff/tests/test_tiff_file.py::test_write_read - SystemError: mmap: resizing not available--no mremap()
FAILED libtiff/tests/test_tiff_image.py::test_rw_rgb - SystemError: mmap: resizing not available--no mremap()
FAILED libtiff/tests/test_tiff_image.py::test_write_read - SystemError: mmap: resizing not available--no mremap()
FAILED libtiff/tests/test_tiff_image.py::test_write_lzw - SystemError: mmap: resizing not available--no mremap()
```

These fail when `.resize` is called on the mmap.

**Comment by djhoese** [comment]
Passing on my fork!

https://github.com/djhoese/pylibtiff/actions/runs/3678864501/jobs/6222637954

**Comment by djhoese** [comment]
@pearu This is ready for review and merge (I can merge if you'd like). The github actions won't show up in this repository until this is merged, but you can see the passing state of them in my fork here:

https://github.com/djhoese/pylibtiff/actions/runs/3679505533

I've added some releasing instructions in a `RELEASING.md` in the root of the repository. Other changes include:

1. Adding `.github/workflows/ci.yaml` directory with tests that run on OSX, Windows, and Linux (ubuntu), an "unstable" environment with the nightly build of numpy, and a "system-libtiff" environment which uses the Ubuntu `apt-get install` version of libtiff instead of conda-forge. These tests *DO NOT* include aarch tests as I don't know of an easy way to implement those in GitHub Actions.
2. Added a CI workflow for deployment of an sdist tarball to PyPI. This is triggered when a GitHub release is created.
3. Added a `.github/dependabot.yaml` which will automatically create PRs (once a month) of any of the tools used in the github CI scripts that have updates.
4. The CI tests include uploading coverage to coveralls.io for tracking test coverage.
5. Updated some tests to use more modern methods of testing like `pytest.mark.parametrize` for testing multiple variations of the same test logic.
6. Had to `pytest.mark.skipif` a couple tests on OSX for the mmap issued mentioned above.
7. Removed Travis and Appveyor configs. You may have to disable these projects on their own sites or disable the integration in the repository's settings.
8. The `ci.yaml` workflow includes a linting check that runs `flake8` to check code style. I fixed quite a few of the initial issues, but there are a lot of docstring issues so in the `setup.cfg` configuration file I told it to just ignore all docstring issues (those starting with `D`).
9. I dropped support for any version of Python <3.8. This is inline with numpy's policy here: https://numpy.org/neps/nep-0029-deprecation_policy.html.

I think that's it.

**Review by pearu — APPROVED** [review]
This looks great! Thank you, @djhoese!
Feel free to merge it.

**Comment by djhoese** [comment]
I just realized without the changes I mentioned in #138 the releasing instructions aren't accurate. I'll need to make another PR regarding automatic versioning. After that, are you ok with me trying to make a release @pearu? It would be a good test of the secret you added. I think since I have PR merge permissions I should have tagging and release permissions too.

**Comment by pearu** [comment]
@djhoese , sure, make a release. I suggest bumping the version to 0.5.0 for this. Thanks!

**Pull request merged**
